hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/marker/WriteMarkers.java

```diff
@@ -86,7 +86,7 @@ public Option<Path> create(String partitionPath, String fileName, IOType type, H
       HoodieTimeline pendingReplaceTimeline = activeTimeline.filterPendingReplaceTimeline();
       // TODO If current is compact or clustering then create marker directly without early conflict detection.
       // Need to support early conflict detection between table service and common writers.
-      if (pendingCompactionTimeline.containsInstant(instantTime) || pendingReplaceTimeline.containsInstant(instantTime)) {
```

Comment:
@jonvex : Wouldn't this cause extra compaction plan read at each writing task level ? Instead, can you see if we can pass this information from the driver ?

Reply:
I reverted my changes to this file. After reading through the code and RFC-56, I don't think early conflict detection is relevant to insert overwrite because new filegroups are also created then. Just checking that it is not a replacecommit is therefore sufficient in this case.

Reply:
@yihua is my understanding correct?